components/Input/styles.css

```diff
@@ -3,25 +3,42 @@
 .Input {
   border: 0;
   padding: 0 0 1em;
+  position: relative;
 }
 
 .Input-label {
+  position: absolute;
   color: #333;
-  display: block;
   font-size: 0.875em;
+  opacity: 0;
   text-transform: uppercase;
+  top: 22px;
 }
 
 .Input-label.is-checkbox {
   font-size: 0.75em;
 }
 
+.Input-label.is-visible {
+  display: block;
+  font-size: 0.7em;
+  opacity: 1;
+  top: 8px;
+  margin-bottom: 0;
+  transition: 0.5s ease top, 0.5s ease opacity;
```

Comment:
move this to the `Input-label` itself so that it animates both up (when content is added) and back down (when content is cleared)